### PR TITLE
Added toolkit of useful components

### PR DESCRIPTION
**Pull request by stefanoborini**
Adds the following components:

- data-table: represents a table of data.
- confirm-dialog: a modal dialog to confirm or reject something
- modal-dialog: as before, but just put in toolkit now

### CODE REVIEW

File: remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        type: String,
+        default: "Box"
+      }
+    },
+    template:
+    '  <div class="row">' +
+    '    <div class="col-md-12">' +
+    '      <div class="box">' +
+    '        <div class="box-header with-border"><slot name="header">{{title}}</slot></div>' +
+    '        <div class="box-body"><slot></slot></div>' +
```
Comment (martinRenou) [inline remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js]: Don't you need a named slot here too ?

Comment (stefanoborini) [inline remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js]: No, I keep it anonymous so everything that is between the component markers is embedded there.

File: remoteappmanager/static/js/admin/vue-components/toolkit/ConfirmDialog.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      },
+      closeCallback: {
+        type: Function,
+        default: function() {}
+      },
+      okCallback: {
+        type: Function,
+        default: function() {}
+      },
+      errorMessage: {
```
Comment (martinRenou) [inline remoteappmanager/static/js/admin/vue-components/toolkit/ConfirmDialog.js]: You never use this errorMessage

File: remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js (hunk 1 above)
Comment (martinRenou) [inline remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js]: Ah ok :)

File: remoteappmanager/static/js/admin/vue-components/toolkit/DataTable.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    props: [
+      "headers", "rows", "globalActions", "rowActions"
+    ],
+    methods: {
+      isBoolean: function(value) {
+        return typeof(value) === "boolean";
+      },
+      buttonClassFromType: function(value) { 
+        var cls = {"btn": true};
+        if (value === undefined) {
```
Comment (martinRenou) [inline remoteappmanager/static/js/admin/vue-components/toolkit/DataTable.js]: With ES6 we will use a default value like in python `function(value = "danger"){...}`

File: remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js (hunk 1 above)
Comment (martinRenou) [inline remoteappmanager/static/js/admin/vue-components/toolkit/AdminLTEBox.js]: Ok :)

File: remoteappmanager/static/js/admin/vue-components/toolkit/ConfirmDialog.js (hunk 2 above)
Comment (stefanoborini) [inline remoteappmanager/static/js/admin/vue-components/toolkit/ConfirmDialog.js]: ! 

thanks.

File: remoteappmanager/static/js/admin/vue-components/toolkit/DataTable.js (hunk 3 above)
Comment (stefanoborini) [inline remoteappmanager/static/js/admin/vue-components/toolkit/DataTable.js]: yes, we need to move to babel so we can code like civilised people.

**Pull request merged**